content/productivity/crawl-planets/plugin.js

```diff
@@ -3,233 +3,328 @@
-    stepperLabel.innerText = 'Max % energy to spend';
-    stepperLabel.style.display = 'block';
-
-    let stepper = document.createElement('input');
-    stepper.type = 'range';
-    stepper.min = '0';
-    stepper.max = '100';
-    stepper.step = '5';
-    stepper.value = `${this.maxEnergyPercent}`;
-    stepper.style.width = '80%';
-    stepper.style.height = '24px';
-
-    let percent = document.createElement('span');
-    percent.innerText = `${stepper.value}%`;
-    percent.style.float = 'right';
-
-    stepper.onchange = (evt) => {
-      percent.innerText = `${evt.target.value}%`;
-      try {
-        this.maxEnergyPercent = parseInt(evt.target.value, 10);
-      } catch (e) {
-        console.error('could not parse energy percent', e);
-      }
+    constructor() {
+        this.planetType = 1;
+        this.minimumEnergyAllowed = false;
+        this.minPlanetLevelTo = 3;
+        this.minPlanetLevelFrom = 0;
+        this.maxPlanetLevelFrom = 3;
+        this.maxEnergyPercent = 85;
     }
-
-    let levelLabel = document.createElement('label');
-    levelLabel.innerText = 'Min. level to capture';
-    levelLabel.style.display = 'block';
-
-    let level = document.createElement('select');
-    level.style.background = 'rgb(8,8,8)';
-    level.style.width = '100%';
-    level.style.marginTop = '10px';
-    level.style.marginBottom = '10px';
-    planetLevels.forEach(lvl => {
-      let opt = document.createElement('option');
-      opt.value = `${lvl}`;
-      opt.innerText = `Level ${lvl}`;
-      level.appendChild(opt);
-    });
-    level.value = `${this.minPlanetLevel}`;
-
-    level.onchange = (evt) => {
-      try {
-        this.minPlanetLevel = parseInt(evt.target.value, 10);
-      } catch (e) {
-        console.error('could not parse planet level', e);
-      }
+    render(container) {
+        container.style.width = '300px';
+
+        let stepperLabel = document.createElement('label');
+        stepperLabel.innerText = 'Max % energy to spend';
+        stepperLabel.style.display = 'block';
+
+        let stepper = document.createElement('input');
+        stepper.type = 'range';
+        stepper.min = '0';
+        stepper.max = '100';
+        stepper.step = '5';
+        stepper.value = `${this.maxEnergyPercent}`;
+        stepper.style.width = '80%';
+        stepper.style.height = '24px';
+
+        let percent = document.createElement('span');
+        percent.innerText = `${stepper.value}%`;
+        percent.style.float = 'right';
+
+        stepper.onchange = (evt) => {
+            percent.innerText = `${evt.target.value}%`;
+            try {
+                this.maxEnergyPercent = parseInt(evt.target.value, 10);
+            } catch (e) {
+                console.error('could not parse energy percent', e);
+            }
+        }
+
+        let minimumEnergyAllowedLabel = document.createElement('label');
+        minimumEnergyAllowedLabel.innerText = 'Allow minimum energy to capture?';
```

Comment:
This isnt great. At worst it should say land with 1 energy instead of 15
At best it shouldnt be a boolean, but a slider set to 15 by default that they can change to whatever

Reply:
Great idea, doing that